Source/AwsCommonRuntimeKit/crt/CBOR.swift

```diff
@@ -0,0 +1,360 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+import AwsCCommon
+import Foundation
+
+/// CBOR Types. These types don't map one-to-one to the CBOR RFC.
+public enum CBORType: Equatable {
+    /// UINT64 type for positive numbers.
+    case uint64(_ value: UInt64)
+    /// INT64 type for negative numbers. If the number is positive, it will be encoded as UINT64 type.
+    case int(_ value: Int64)
+    /// Double type. It might be encoded as an integer if possible without loss of precision. Half-precision floats are not supported.
+    case double(_ value: Double)
+    /// Bytes type for binary data
+    case bytes(_ value: Data)
+    /// Text type for utf-8 encoded strings
+    case text(_ value: String)
+    /// Array type
+    case array(_ value: [CBORType])
+    /// Map type
+    case map(_ value: [String: CBORType])
```

Comment:
Please validate, asserting that map keys will always be string. For indefinite map, it's up to the users.